### PR TITLE
[ticket/15055] Add appveyor file to allow running tests on appveyor as well

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15055

### CODE REVIEW

**Comment by Nicofuma** [comment]
You should add appveyor to phpbb/phpbb so we can see the result :)

**Comment by marc1706** [comment]
Was hoping to get this to work first. Will add it once that's the case.

**Comment by Noxwizard** [comment]
The connection string needs updating on the phpBB side. Add the following:
`'CharacterSet' => 'UTF-8'`
to the connection details array: https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/db/driver/mssqlnative.php#L51

The default for `sqlsrv_connect()` is [SQLSRV_ENC_CHAR](https://docs.microsoft.com/en-us/sql/connect/php/connection-options) while the default for PDO connections is [PDO::SQLSRV_ENCODING_UTF8](https://docs.microsoft.com/en-us/sql/connect/php/pdostatement-setattribute#remarks).

File: tests/console/user/base.php [hunk 1]
```diff
@@ -34,6 +34,11 @@ public function setUp()
 	{
 		global $auth, $db, $cache, $config, $user, $phpbb_dispatcher, $phpbb_container, $phpbb_root_path, $phpEx;
 
+		if (strtolower(substr(PHP_OS, 0, 5)) !== 'linux')
+		{
+			$this->markTestSkipped('Unable to test console feature on OS other than Linux.');
```
Comment (Nicofuma) [inline tests/console/user/base.php]: actually it should work on windows I think

File: tests/dbal/db_tools_test.php [hunk 2]
```diff
@@ -455,7 +462,10 @@ public function test_create_index_with_long_name()
 		// Index name has > 30 chars - that should not be possible.
 		$too_long_index_name = str_repeat('i', 31);
 		$this->assertFalse($this->tools->sql_index_exists('prefix_table_name', $too_long_index_name));
-		$this->setExpectedTriggerError(E_USER_ERROR);
+		if (strpos($this->tools->sql_layer, 'mssql') === false)
+		{
+			$this->setExpectedTriggerError(E_USER_ERROR);
```
Comment (Nicofuma) [inline tests/dbal/db_tools_test.php]: why we don't have an error on mssql? it's worth a comment imo

File: tests/test_framework/phpbb_database_test_case.php [hunk 3]
```diff
@@ -145,6 +145,69 @@ protected function database_synchronisation($table_column_map)
 	public function createXMLDataSet($path)
 	{
 		$this->fixture_xml_data = parent::createXMLDataSet($path);
+		if (strpos($this->get_database_config()['dbms'], 'mssql') !== false)
```
Comment (Nicofuma) [inline tests/test_framework/phpbb_database_test_case.php]: could you add a comment explaining what is being done and why please?

File: tests/tree/nestedset_forum_base.php [hunk 4]
```diff
@@ -100,13 +100,23 @@ protected function set_up_forums()
 		}
 		else
 		{
+			if (strpos($this->db->sql_layer, 'mssql') !== false)
```
Comment (Nicofuma) [inline tests/tree/nestedset_forum_base.php]: again, what and why please

Comment (marc1706) [reply tests/tree/nestedset_forum_base.php]: Added comment

**Pull request merged**